### PR TITLE
Use a configured base URL

### PR DESCRIPTION
**Pull request by APIYOJENNIFER**
Issue #688 
- Use a configured base URL to be able to test API endpoints in different environments

### CODE REVIEW

File: frontend/src/utils/api.ts [hunk 1]
```diff
@@ -1,31 +1,33 @@
+import config from '../config';
```
Comment (ThatcherK) [inline frontend/src/utils/api.ts]: I don't expect any changes in this file

File: frontend/src/config.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+const config: { BASE_URL: string } = {
```
Comment (ThatcherK) [inline frontend/src/config.ts]: This is not an environment variable..
I expect an axios config, and then the config takes in an environment variable as the baseUrl

**Review by ThatcherK — CHANGES_REQUESTED** [review]

File: frontend/src/config.ts [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import axios from 'axios';
+
+const axiosConfig = axios.create({
```
Comment (ThatcherK) [inline frontend/src/config.ts]: This config does not seem to be referenced anywhere else apart from this file. So I doubt it's working...
Please see if [this](https://bobbyhadz.com/blog/axios-change-base-url#:~:text=%23%20Setting%20the%20axios%20Base%20URL%20by%20creating%20an%20instance,'%2C%20%7D)%3B%20placeholderApi%20.) resource helps

**Review by ThatcherK — CHANGES_REQUESTED** [review]

**Review by ThatcherK — CHANGES_REQUESTED** [review]
Much as this might work, I would rather you add the baseUrl in the index.js file which doesnt require you to add it to all places where axios is called. Less code. Sometrhing similar to this 
`axios.defaults.baseURL` that sets it as a global baseUrl for all axios calls. Also make sure you r environment variable is actually being read as what you have set...

File: frontend/src/index.ts [hunk 4]
```diff
@@ -1 +1,4 @@
+import axios from 'axios';
 import './App';
+
+axios.defaults.baseURL = process.env.BASE_URL;
```
Comment (ThatcherK) [inline frontend/src/index.ts]: As I sort my testing issues,move this to the App.ts file, I've seen most configs are initialised there. Let's leave the index file as clean as we found it!.. Also test that it still works. Also change the environment variable name to API_BASE_URL( its abit more descriptive)

**Review by ThatcherK — CHANGES_REQUESTED** [review]

File: frontend/src/index.ts (hunk 4 above)
Comment (ThatcherK) [reply frontend/src/index.ts]: You will also need to add it to the docker_compose files. Check the environment section of the web service.
Then also add it to the readme as a variable that needs to be set

**Comment by ThatcherK** [comment]
@davidebukali can you test that this works, am unable to test on my end

**Review by ThatcherK — COMMENTED** [review]
Waiting to properly test before merging

**Comment by davidebukali** [comment]
@ThatcherK this works on my side with api base url set from .env file

**Review by ThatcherK — APPROVED** [review]

**Pull request merged**